statefulset added

drohnow opened the pull request:
@diamonwiggins I accidentally deleted the previous draft PR so here is a new one.

- Added _statefulset.tpl
- Add documentation in values.yaml

drohnow commented:
Hi @diamonwiggins I thought that I would give the serviceName conditional a crack.  Please let me know what you think.

diamonwiggins commented on charts/replicated-library/templates/lib/_statefulset.tpl:
```diff
@@ -0,0 +1,72 @@
+  selector:
+    matchLabels:
+      {{- include "replicated-library.labels.selectorLabels" . | nindent 6 }}
+  {{- if $values.serviceName}}
```
Checkout - https://helm.sh/docs/chart_template_guide/function_list/#default. Similiar to line 27 in this file it can probably keep the code a bit simpler. Also, instead of `...names.fullname` we should be using the appName of the current app instead.

drohnow replied:
{{- $serviceName := default (include "replicated-library.names.appname" .) ($values.serviceName) }}

diamonwiggins commented on charts/replicated-library/templates/lib/_statefulset.tpl:
```diff
@@ -0,0 +1,72 @@
+{{/*
+This template serves as the blueprint for the StatefulSet objects that are created
+within the common library.
```
"common library" should be `replicated library`

drohnow replied:
resolved

diamonwiggins commented:
@drohnow Addressing the feedback on the conditional as well as fixing the linting errors should have this one ready for merge. Also don't forget to bump the chart version, update the changelog, and re-generate the readme. Instructions on re-generating the readme are in the readme

drohnow commented:
> @drohnow Addressing the feedback on the conditional as well as fixing the linting errors should have this one ready for merge. Also don't forget to bump the chart version, update the changelog, and re-generate the readme. Instructions on re-generating the readme are in the readme



> 



> 
- I bumped the chart version to 4.0 -- is this what it should be?
- Do I directly README_CHANGELOG.md.gotmpl (if so that it what I did).
- regen'd  README.
- Linting errors: I ran helm lint and did not get any errors.

drohnow commented:
@diamonwiggins fixed linting errors.

diamonwiggins commented on charts/replicated-library/Chart.yaml:
```diff
@@ -2,7 +2,9 @@ apiVersion: v2
 name: replicated-library
 description: Replicated library chart
 type: library
+
```
Any reason for these extra spaces here?

diamonwiggins commented on charts/replicated-library/README.md:
```diff
@@ -1,6 +1,23 @@
+=======
+>>>>>>> 00be428 (Update version to 0.4.0 (#25))
+<<<<<<< HEAD
+![Version: 0.5.0](https://img.shields.io/badge/Version-0.5.0-informational?style=flat-square) ![Type: library](https://img.shields.io/badge/Type-library-informational?style=flat-square)
```
not sure what happened here? Git foo gone wrong?

diamonwiggins commented on charts/replicated-library/README.md:
```diff
@@ -37,7 +54,15 @@ Include the chart as a dependency in your `Chart.yaml`
 - name: replicated-library
   repository: https://replicatedhq.github.io/helm-charts
+<<<<<<< HEAD
+<<<<<<< HEAD
```
not sure what happened here? Git foo gone wrong?

diamonwiggins commented on charts/replicated-library/README.md:
```diff
@@ -180,6 +208,10 @@ All notable changes to this library Helm chart will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+<<<<<<< HEAD
```
not sure what happened here? Git foo gone wrong?

diamonwiggins commented on charts/replicated-library/templates/lib/_statefulset.tpl:
```diff
@@ -0,0 +1,69 @@
+    matchLabels:
+      {{- include "replicated-library.labels.selectorLabels" . | nindent 6 }}
+  {{- $serviceName := default (include "replicated-library.names.appname" .) ($values.serviceName) }}
+  serviceName:  {{ $serviceName }}
```
i think we can skip the variable here since we're only using this in one spot. just use the `default` function right on `serviceName`. `serviceName: {{ default ...`

diamonwiggins reviewed (changes requested):
a few things. also you need to bump the version to `0.5.0` in the Chart.yaml

chris-sanders commented:
> also you need to bump the version to `0.5.0` in the Chart.yaml

You'll want to add 0.5.0 to the Changelog (just follow the existing formatting) and re-run helm-docs after you've done that as well. You may have know that, but I missed it the first time I was doing it.

drohnow commented:
> > also you need to bump the version to `0.5.0` in the Chart.yaml
> 
> You'll want to add 0.5.0 to the Changelog (just follow the existing formatting) and re-run helm-docs after you've done that as well. You may have know that, but I missed it the first time I was doing it.
> 
> Oh that reminds me @diamonwiggins I guess while doing this just remove the "Unreleased" section of the changelog? I added it following the listed format but i believe we never intended to land unreleased changes so it's not relevant anymore? If @drohnow you could just take that bit out while you're adding in 0.5.0

Just saw this and removed 'unreleased' in new draft PR.

chris-sanders commented:
As far as I can tell this is superseded by #30
